frontend/src/components/UserProfile/UserProfileEditPassword.tsx

```diff
@@ -0,0 +1,253 @@
+      message: 'Password minimum length is 6',
+    },
+    maxLength: {
+      value: 255,
+      message: 'Email field max character up to 255 only.',
+    },
+  },
+};
+
+interface Props {
+  userUpdateProfileDetails: Function;
+}
+
+const _UserProfileEditPassword = (props: Props) => {
+  const [cookies] = useCookies();
+
+  const [password, setPassword] = useState('');
+  const [new_password, setNew_password] = useState('');
+  const [new_password_confirmation, setNew_password_confirmation] =
+    useState('');
+
+  const [error, setError] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [msg, setMsg] = useState('');
+
+  const [showPassword, setShowPassword] = useState(false);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<ProfileEditPasswordInput>();
+
+  const onSubmit = (data: ProfileEditPasswordInput) => {
+    setIsLoading(true);
+    setError(false);
+    setSuccess(false);
+    let profilePasswordData = {
+      token: cookies.token,
+      user_id: cookies.user.id,
+      password: data.password,
+      new_password: data.new_password,
+      new_password_confirmation: data.new_password_confirmation,
+      callback: () => {
+        setSuccess(true);
+        setMsg('Password Changed Successfully');
+      },
+      errorCallback: (err: AxiosError) => {
+        setError(true);
+        let errordata = err.response?.data;
+        setMsg(errordata?.error?.message);
+      },
+      finallyCallback: () => {
+        setIsLoading(false);
+      },
+    };
+
+    userUpdatePassword(profilePasswordData);
+  };
+
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const handleMouseDownPassword = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+  };
+
+  return (
+    <Box
+      component="form"
+      onSubmit={handleSubmit(onSubmit)}
+      sx={{
+        '& .MuiTextField-root': { m: 1, width: '25ch' },
+      }}
+      noValidate
+      autoComplete="off"
+    >
+      <div>
+        <TextField
+          {...register('password', formValidation.password)}
+          type={showPassword ? 'text' : 'password'}
+          error={Boolean(errors?.password)}
+          label="Current Password"
+          value={password}
+          onChange={(e) => setPassword(e.target.value)}
+          variant="filled"
+          helperText={errors?.password?.message}
+          InputProps={{
+            endAdornment: (
+              <InputAdornment position="end">
+                <IconButton
+                  aria-label="toggle password visibility"
+                  onClick={handleClickShowPassword}
+                  onMouseDown={handleMouseDownPassword}
+                  edge="end"
+                >
+                  {showPassword ? <VisibilityOff /> : <Visibility />}
+                </IconButton>
+              </InputAdornment>
```

Comment:
create new component for this one

Reply:
its already a component sir from  @MUI, maybe just use a variable so that it wont repeat?

Reply:
Yes you can, since it is repeating